Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFloor.cs

```diff
@@ -240,17 +240,16 @@ private ICurve GetFlattenedCurve(ICurve curve, double z)
 
         case OG.Polyline poly:
           var polylinePonts = new List<double>();
-          var originalPolylinePoints = poly.ToList();
+          var originalPolylinePoints = poly.points;
           var polyConversionFactor = z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, poly.units);
           for (var i = 0; i < originalPolylinePoints.Count; i++)
           {
-            if (i % 3 == 2)
-              polylinePonts.Add(z * polyConversionFactor);
-            else
-              polylinePonts.Add(originalPolylinePoints[i]);
+            polylinePonts.Add(originalPolylinePoints[i].x);
+            polylinePonts.Add(originalPolylinePoints[i].y);
+            polylinePonts.Add(originalPolylinePoints[i].z * polyConversionFactor);
```

Comment:
This last line should be ```polylinePonts.Add(z * polyConversionFactor);```. We're changing all z values to the same value, not scaling each individual value.